tests/Database/DatabaseEloquentBelongsToManyTest.php

```diff
@@ -116,7 +116,7 @@ public function testModelsAreProperlyMatchedToParents()
         $this->assertEquals(2, $models[1]->foo[0]->pivot->user_id);
         $this->assertEquals(2, $models[1]->foo[1]->pivot->user_id);
         $this->assertEquals(2, count($models[1]->foo));
-        $this->assertEquals(0, count($models[2]->foo));
+        $this->assertEquals(0, count((array) $models[2]->foo));
```

Comment:
It seems strange that you need to use `(array)` on the results which are `0` to make them pass, but not on the results that are not `0`.

If you dont know what the result is going to be before using `count()` - that could be a problem.

Kinds of feels like we are hiding a bigger issue? Or at the minimum we are not being consistent in the tests...